### ISSUE TITLE
Poor Logits when fine tuning RTDETR

### ISSUE THREAD

**Issue by sodaJar** [lyuwenyu/RT-DETR#664]
When I fine tune RT-DETR or RT-DETR V2 on a custom dataset (with a different nr of classes), the logits from the last decoder layer (layer 5) for the correct class is a negative number (e.g. -5) and are only slightly bigger than incorrect classes (e.g. -6), leading to low confidence scores. However, intermediate logits from layers 0 to 4 are all fine (~3 vs ~-6). Is this a known issue?

```
with torch.no_grad():
    outputs = model(**inputs)
outputs.logits = outputs.intermediate_logits[:, -2]
```
the last line makes the post processor use logits from the second last layer instead of the last, which fixes the problem of poor logits.

在自己数据集上微调后的 RT-DETR 或 RT-DETR V2输出的正确类别的logits值是相对较大的负数，sigmoid后就是个很小的score ~0.001，但是第0-4 层的outputs.intermediate_logits都是正常的，正确class的logits值喂给sigmoid出来的score就是0.9。有人知道这是什么原因吗？？？
